modules/images/webp-uploads/load.php

```diff
@@ -4,41 +4,290 @@
+		return new WP_Error( 'image_mime_type_invalid', __( 'The provided mime type is not allowed.', 'performance-lab' ) );
+	}
+
+	if ( ! wp_image_editor_supports( array( 'mime_type' => $mime ) ) ) {
+		return new WP_Error( 'image_mime_type_not_supported', __( 'The provided mime type is not supported.', 'performance-lab' ) );
 	}
 
-	$output_format['image/jpeg'] = 'image/webp';
+	$extension = explode( '|', $allowed_mimes[ $mime ] );
+	$extension = reset( $extension );
 
-	return $output_format;
+	$width  = null;
+	$height = null;
+	$crop   = false;
+
+	if ( array_key_exists( 'width', $metadata['sizes'][ $size ] ) ) {
+		$width = $metadata['sizes'][ $size ]['width'];
+	} elseif ( array_key_exists( 'width', $sizes[ $size ] ) ) {
+		$width = $sizes[ $size ];
+	}
+
+	if ( array_key_exists( 'height', $metadata['sizes'][ $size ] ) ) {
+		$height = $metadata['sizes'][ $size ]['height'];
+	} elseif ( array_key_exists( 'width', $sizes[ $size ] ) ) {
+		$height = $sizes[ $size ];
+	}
+
+	if ( array_key_exists( 'crop', $sizes[ $size ] ) ) {
+		$crop = (bool) $sizes[ $size ]['crop'];
+	}
+
+	$editor->resize( $width, $height, $crop );
+	$filename = $editor->generate_filename( null, null, $extension );
+	$filename = preg_replace( '/-(scaled|rotated|imagifyresized)/', '', $filename );
+	$image    = $editor->save( $filename, $mime );
+
+	if ( is_wp_error( $image ) ) {
+		return $image;
+	}
+
+	if ( empty( $image['file'] ) ) {
+		return new WP_Error( 'image_file_not_present', __( 'The file key is not present on the image data', 'performance-lab' ) );
+	}
+
+	if ( ! isset( $metadata['sizes'][ $size ]['sources'] ) || ! is_array( $metadata['sizes'][ $size ]['sources'] ) ) {
+		$metadata['sizes'][ $size ]['sources'] = array();
+	}
+
+	$metadata['sizes'][ $size ]['sources'][ $mime ] = array(
+		'file'     => $image['file'],
+		'filesize' => array_key_exists( 'path', $image ) ? filesize( $image['path'] ) : 0,
+	);
+
+	return wp_update_attachment_metadata( $attachment_id, $metadata );
+}
+
+add_action( 'webp_uploads_create_image', 'webp_uploads_generate_image_size', 10, 3 );
+
+/**
+ * Returns an array with the list of valid mime types that a specific mime type can be converted into it,
+ * for example an image/jpeg can be converted into an image/webp.
+ *
+ * @since n.e.x.t
+ *
+ * @return array<string, array<string>> An array of valid mime types, where the key is the mime type and the value is the extension type.
+ */
+function webp_uploads_get_supported_image_mime_transforms() {
+	$image_mime_transforms = array(
+		'image/jpeg' => array( 'image/webp' ),
+		'image/webp' => array( 'image/jpeg' ),
+	);
+
+	/**
+	 * Filter to allow the definition of a custom mime types, in which a defined mime type
+	 * can be transformed and provide a wide range of mime types.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array<string, array<string>> An array with the valid mime transforms.
+	 *
+	 * @return array<string, array<string>> An array with the valid mime transformation
```

Comment:
This is missing the variable name, which needs to be included in filter param docs. What it does _not_ need is the `@return` doc, filters don't have that. Also arrays in WP core can simply be documented as `array`, or if they are a list of just one data type like `[]string`.
```suggestion
	 * @param array $image_mime_transforms A map with the valid mime transforms.
```